src/Transports.AspNetCore/HttpGetValidationRule.cs

```diff
@@ -0,0 +1,23 @@
+#nullable enable
+
+using GraphQL.Server.Transports.AspNetCore.Errors;
+using GraphQL.Validation;
+using GraphQLParser.AST;
+
+namespace GraphQL.Server.Transports.AspNetCore;
+
+/// <summary>
+/// Validates that HTTP GET requests only execute queries; not mutations or subscriptions.
+/// </summary>
+public sealed class HttpGetValidationRule : IValidationRule
```

Comment:
👍 I like these new rules. They fit well into the design.

Reply:
But what in case of batched requests? There will be partial execution.

Reply:
It is no different than any other validation error that can occur for one of the multiple executions, such as a malformed query that cannot even be parsed. It is not the fault of the middleware if one of the requests is invalid.

Reply:
The reason I added the option for parallel execution is because I envisioned batching as just a manner to batch some independent requests together. In the same manner, here, the fact that one request is invalid has no effect on other requests. 

However I can see that perhaps someone specifically wants or intended batched requests to execute in series. That’s why an option was provided. 

For my own needs, I added a “query” field to my mutation type that returns the query type. In this manner I can execute queries after a mutation as part of the same database transaction, and the entire database transaction will roll back if any part fails. In addition, within GraphQL you can request the same field multiple times with aliases. So for my own code, I can execute a mutation which effectively lets me execute any combination of queries or mutations in a prescribed order within a single database transaction. Now, that’s just my implementation. Others could probably achieve the same result in batched queries by overriding the correct methods of the middleware to start and commit the transaction, and roll it back in the case of failure of any query.

Reply:
True. Makes sense.